### PR TITLE
fix/disambiguate-property-meaning, make JSON valid in basic-javascript.json

### LINKED ISSUE #8955: Record Collection challenge has incorrect JSON syntax
The task says: "You are given a JSON object representing..."

Start of the example code given at the start:

```
var collection = {
    2548: {
      album: "Slippery When Wet",
      artist: "Bon Jovi",
      tracks: [ 
        "Let It Rock", 
        "You Give Love a Bad Name" 
```

Quote on JSON objects from MDN:

> JSON permits **only property definition using _"property": value syntax_**.  The property name must be double-quoted, and the definition cannot be a shorthand.

Source: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Object_initializer#Object_literal_notation_vs_JSON

Hence, the statement that it is a JSON object is wrong - it is JS object as properties are not written as strings. Personally, it made me confused because there is another task:
https://www.freecodecamp.com/challenges/profile-lookup
Where properties are written correctly with quotes. As a result, it made me think that both options are the correct declaration of JSOn objects.

I hope that can be fixed.

Thanks,
Aleks

### PR DESCRIPTION
**Pull request by LindsayDenotes**
#### Pre-Submission Checklist

- [x ] Your pull request targets the `staging` branch of FreeCodeCamp.
- [x ] Branch starts with either `fix/`, `feature/`, or `translate/` (e.g. `fix/signin-issue`)
- [x ] You have only one commit (if not, [squash](https://github.com/FreeCodeCamp/FreeCodeCamp/wiki/git-rebase#squashing-multiple-commits-into-one) them into one commit).
- [x ] All new and existing tests pass the command `npm run test-challenges`. Use `git commit --amend` to amend any fixes.
#### Type of Change

- [ ] Small bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds new functionality)
- [x ] Breaking change (fix or feature that would change existing functionality)
- [ ] Add new translation (feature adding new translations)
#### Checklist:

- [x ] Tested changes locally.
- [x ] Closes currently open issue: Closes #8955
#### Description

… Notation JSON: I revised the explanations from line 4322 to line 4328 in order to define JSON and provide brief explanations. I modified the code on line 4343 by adding quotation marks around the boolean value in order to format the object as valid JSON. This did not break the functionality of the existing published code (as of 6:15 PM on June 11, 2016) when I tested that, but please double check that. Further, for the title Record Collection: I revised the instructions from line 4502 to line 4506 in order to disambiguate the meaning of key from the meaning of property. I added to the notes from line 4510 to line 4511 to provide a link to MDN which explains the push method. I modified the code from line 4517 to line 4538 by adding quotation marks around the keys in order to format it as valid JSON. This change at first DID break the functionality of the existing published code when I tested it. So I changed line 4542 from var collectionCopy = JSON.parse(JSON.stringify(collection)); to var collectionCopy = JSON.stringify(collection);.

### CODE REVIEW

File: seed/challenges/01-front-end-development-certification/basic-javascript.json [hunk 1]
```diff
@@ -4319,11 +4319,13 @@
...
+        "JavaScript Object Notation or <code>JSON</code> is a data interchange format used to store data (source: json.org). JSON is heterogeneous, meaning it permits some properties to have values that other properties do not have.",
+        "A property is the part of an object that associates a key (either a String value or a Symbol value) and a value (source: [http://www.ecma-international.org/ecma-262/6.0/#sec-property](url)). So, a property consists of a key - value pair. (source: [http://spacetelescope.github.io/understanding-json-schema/reference/object.html#properties](url)).",
+        "Like JavaScript Objects, JSON is flexible in that it permits <dfn>Data Structures</dfn> with arbitrary combinations of <dfn>strings</dfn>, <dfn>arrays</dfn>, and <dfn>objects</dfn>, but JSON is not allowed to contain booleans or numbers unless they are wrapped in quotation marks, which makes them strings instead."
```
Comment (erictleung) [inline seed/challenges/01-front-end-development-certification/basic-javascript.json]: You're missing a comma at the end of this line.

**Comment by robjloranger** [comment]
@LindsayBrown81 As @erictleung mention on gitter you need to use html links instead of markdown.
Also you have two commits, you need to squash them. Perhaps after you make your next changes, then squash all three before you push again.

Here is the Travis CI error:
``` error [SyntaxError: /home/travis/build/FreeCodeCamp/FreeCodeCamp/seed/challenges/01-front-end-development-certification/basic-javascript.json: Unexpected token ]] ```
Looks like a syntax error, somewhere there is an extra or erroneous ``` ] ```

**Comment by LindsayBrown81** [comment]
Thank you so much. Will do on Wednesday.

**Comment by robjloranger** [comment]
@LindsayBrown81 You still need to squash your commits.
[Git Squash on FCC Wiki](https://github.com/FreeCodeCamp/FreeCodeCamp/wiki/Git-Squash)

```
git rebase -i HEAD~4
```
Then edit the file changing all the ```pick```'s to ```squash```'s **except** the one that describes your fix, that one stays as ```pick```

**Comment by LindsayBrown81** [comment]
Rob,
 
Thank you. As this was my first attempt at squashing, I made a mistake and asked for help on https://gitter.im/FreeCodeCamp/HelpContributors at 3:58. 
 
I pasted my gitter message below which outlines the problem I have created:
@erictleung I made a mistake and need advice on how not to screw up anyone else's PRs. I managed to squash MY commits into one commit; however, when doing git rebase on my own fix/ branch, I reset the HEAD back too many index positions. I assumed that if I entered in a number that was greater than the number of commits I pushed to my own branch, then I would receive some type of error. What it actually did was include commits from 3 other developers that did their commits either weeks before I ever created my own branch or to a entirely different challenge file. I did not make any edits to their code at all, which the diffs reflect. So now I have a PR that says I want to merge 4 commits into FreeCodeCamp: staging. How do I correct this? 
 Please advise what I should do next to remove those other three developers' commits from my branch and/or PR.
 Thank you again,
 
Lindsay 
 
--------- Original Message --------- Subject: Re: [FreeCodeCamp/FreeCodeCamp] fix/disambiguate-property-meaning, make JSON valid in basic-javascript.json (#9096)

**Comment by LindsayBrown81** [comment]
Hi Rob,
 
When I did git rebase -i HEAD~4 and then in VI, changed all the pick's to squash's except the one that describes my fix, I got the following error: 
 
Cannot 'squash' without a previous commit
 
I really got myself into a pickle. The three commits I tried to squash are ones I did not commit myself. Other developers did them. 
 
 
My best guess at what I should do next is:
git reset --soft HEAD~1
git rm -r
git push -f
I believe that should put me back to having just my own three commits to squash (includes the two already pushed and the last set of changes I made this afternoon). Does this approach sound good?
 
Thanks again,
Lindsay
 
 
 
--------- Original Message --------- Subject: Re: [FreeCodeCamp/FreeCodeCamp] fix/disambiguate-property-meaning, make JSON valid in basic-javascript.json (#9096)

**Comment by robjloranger** [comment]
@LindsayBrown81 If you haven't done anything since the rebase, i would try ```git reset --hard ORIG_HEAD``` as git saves your HEAD to ORIG_HEAD when doing rebase, merge and reset

**Comment by LindsayBrown81** [comment]
If I use HARD, will that wipe out my local changes I made to basic-javascript.json?

**Comment by robjloranger** [comment]
Oh yes, sorry. Use --soft
Oops

Rob Loranger
250-589-4806
Saanich, B.C.
www.robloranger.ca

**Comment by robjloranger** [comment]
@LindsayBrown81 Looks good :+1:

**Comment by LindsayBrown81** [comment]
Thank you. What an adventure! Someone might want to double check how I reworded lines 4505-4508. I haven't completed that challenge myself because I (and other campers) found the wording so confusing.

**Comment by atjonathan** [comment]
@LindsayBrown81, you currently have [merge conflicts](https://github.com/FreeCodeCamp/FreeCodeCamp/wiki/Git-Rebase#preventing-merge-conflicts). Please checkout the tutorial linked :smile:

**Comment by LindsayBrown81** [comment]
@atjonathan Okay. I ran git pull --rebase upstream staging. My terminal gave me the message below so at this point, it looks like I need to fix a patch. Would you please point me to a helpful web page about how to do that? 
...
Patch failed at 0001 fix/disambiguate-property-meaning, make JSON valid in basic-javascript.json #9096
The copy of the patch that failed is found in: .git/rebase-apply/patch

When you have resolved this problem, run "git rebase --continue".
If you prefer to skip this patch, run "git rebase --skip" instead.
To check out the original branch and stop rebasing, run "git rebase --abort".
...
I also double checked to make sure my remotes are set correctly. They look right to me (see below), but please let me know if something is amiss. Thank you! 

Lindsays-MacBook-Pro-2:FreeCodeCamp lindsaybrown$ git remote -v
origin	https://github.com/LindsayBrown81/FreeCodeCamp.git (fetch)
origin	https://github.com/LindsayBrown81/FreeCodeCamp.git (push)
upstream	https://github.com/FreeCodeCamp/FreeCodeCamp.git (fetch)
upstream	https://github.com/FreeCodeCamp/FreeCodeCamp.git (push)

**Comment by atjonathan** [comment]
@LindsayBrown81, those remotes are fine. http://stackoverflow.com/questions/161813/fix-merge-conflicts-in-git has some useful pointers!

**Comment by atjonathan** [comment]
#### **LGTM :+1:**
Thanks for contributing :tada:

**Pull request merged**